platform/ios/src/MGLMapView.mm

```diff
@@ -1508,18 +1573,36 @@ - (void)handleDoubleTapGesture:(UITapGestureRecognizer *)doubleTap
 
     if (doubleTap.state == UIGestureRecognizerStateEnded)
     {
-        [self trackGestureEvent:MGLEventGestureDoubleTap forRecognizer:doubleTap];
-        CGPoint gesturePoint = [self anchorPointForGesture:doubleTap];
-
-        mbgl::ScreenCoordinate center(gesturePoint.x, gesturePoint.y);
-        _mbglMap->scaleBy(2, center, MGLDurationInSecondsFromTimeInterval(MGLAnimationDuration));
-
-        __weak MGLMapView *weakSelf = self;
-
-        [self animateWithDelay:MGLAnimationDuration animations:^
+        MGLMapCamera *oldCamera = self.camera;
+        
+        mbgl::EdgeInsets padding = MGLEdgeInsetsFromNSEdgeInsets(self.contentInset);
```

Comment:
nitpick: these several lines about getting padding, camera options, setting a zoom adjustment, clamping, and getting a `toCamera` from camera options are repeated in the `handleDoubleTapGesture:`, `handleTwoFingerTapGesture:` and `handleQuickZoomGesture:` implementations. This is similar to what we had before where the previous (less complex because no estimate) logic was repeated, but this PR could be a chance to refactor this into a private utility method. I think anything we can do to make the gesture handling more readable and easier to refactor in the future would be 👍 so that is why I mention this.